components/twitter_developer_app/sources/new-message/new-message.js

```diff
@@ -0,0 +1,76 @@
+const common = require("../common");
+
+module.exports = {
+  ...common,
+  key: "twitter-new-message",
+  name: "New Message",
+  description: "Emit an event each time the authenticated user receives a Direct Message",
+  version: "0.0.1",
+  dedupe: "unique",
+  hooks: {
+    async deploy() {
+      const accountId = await this.twitter_developer_app.getAccountId();
+      this._setAccountId(accountId);
+    },
+  },
+  methods: {
+    _getAccountId() {
+      return this.db.get("accountId");
+    },
+    _setAccountId(accountId) {
+      this.db.set("accountId", accountId);
+    },
+    _getNextCursor() {
+      return this.db.get("nextCursor");
+    },
+    _setNextCursor(nextCursor) {
+      this.db.set("nextCursor", nextCursor);
+    },
+    _isRelevant(directMessage) {
+      return (
+        directMessage.type === "message_create" &&
+        directMessage.message_create.sender_id != this._getAccountId()
```

Comment:
Will the `message_create` intermediate object ever not exist?

Reply:
It will if the type of the message is `message_create`, but I'll add a check just to be safe